stix_shifter_modules/paloalto/stix_transmission/results_connector.py

```diff
@@ -0,0 +1,248 @@
+from stix_shifter_utils.modules.base.stix_transmission.base_results_connector import BaseResultsConnector
+from stix_shifter_utils.utils.error_response import ErrorResponder
+from stix_shifter_utils.utils import logger
+import json
+from requests.exceptions import ConnectionError
+from .response_mapper import ResponseMapper
+from os import path
+
+TO_STIX_PATH = "../stix_translation/json/to_stix_map.json"
+CONFIG_MAP_PATH = "../stix_translation/json/config_map.json"
+
+
+class InvalidQueryException(Exception):
+    pass
+
+
+class StreamNotImplementedException(Exception):
+    pass
+
+
+class ResultsConnector(BaseResultsConnector):
+
+    def __init__(self, api_client):
+        self.api_client = api_client
+        self.logger = logger.set_logger(__name__)
+
+    @staticmethod
+    def load_json(rel_path_of_file):
+        """
+        Consumes a json file and returns a dictionary
+        :param rel_path_of_file: str
+        :return: dict
+        """
+        _json_path = path.dirname(path.realpath(__file__)) + "/" + rel_path_of_file
+        if path.exists(_json_path):
+            with open(_json_path, encoding='utf-8') as f_obj:
+                return json.load(f_obj)
+        raise FileNotFoundError
+
+    def create_results_connection(self, search_id, offset, length):
+        """
+        Fetching the results using search id, offset and length
+        :param search_id: str, search id generated in transmit query
+        :param offset: str, offset value
+        :param length: str, length value
+        :return: dict
+        """
+
+        response_dict = {}
+        return_obj = {}
+        results = []
+        response_wrapper = None
+        try:
+            min_range = int(offset)
+            max_range = int(offset) + int(length)
+
+            # Grab the response, extract the response code, and convert it to readable json
+            response_wrapper = self.api_client.get_search_results(search_id)
+            response_code = response_wrapper.code
+            response_text = json.loads(response_wrapper.read().decode('utf-8'))
+            if response_code != 200:
+                return_obj = ResponseMapper().status_code_mapping(response_code, response_text)
+            else:
+                if 'status' in response_text['reply'].keys() and response_text['reply']['status'] \
+                        in ('SUCCESS', 'PARTIAL_SUCCESS'):
+                    if 'data' in response_text['reply']['results'].keys() and \
+                            response_text['reply']['number_of_results'] > 0:
+                        results = ResultsConnector.format_results_data(response_text['reply']['results']['data'])
+                    elif 'stream_id' in response_text['reply']['results'].keys():
```

Comment:
would there be any case where both `data` and `stream_id` return by the api?

Reply:
No. The API will return either the data or stream id. 
Reference : https://docs.paloaltonetworks.com/cortex/cortex-xdr/cortex-xdr-api/cortex-xdr-apis/xql-apis/get-xql-query-results.html#id02a1cd3a-b305-4be6-834e-51373c293e56